### PR TITLE
OpenReadWriteFile: add test

### PR DESCRIPTION
**Pull request by MichaelMure**
Address https://github.com/ipld/go-car/pull/323#issuecomment-1200823708

### CODE REVIEW

File: v2/blockstore/readwrite_test.go [hunk 1]
```diff
@@ -945,6 +945,38 @@ func TestReadWrite_ReWritingCARv1WithIdentityCidIsIdenticalToOriginalWithOptions
...
+	_, err = bs.Get(ctx, roots[0])
+	require.NoError(t, err)
+	_, err = bs.GetSize(ctx, roots[0])
+	require.NoError(t, err)
+
+	err = bs.Finalize()
+	require.NoError(t, err)
+
+	_, err = f.Seek(0, 0)
+	require.NoError(t, err) // file should not be closed, let the caller do it
```
Comment (masih) [inline v2/blockstore/readwrite_test.go:977]: The test is the caller, and therefore should close the file. No?

Comment (MichaelMure) [reply v2/blockstore/readwrite_test.go:977]: at least windows seems to agree with you, I'll add that

File: v2/index/index.go [hunk 2]
```diff
@@ -102,6 +102,13 @@ func GetFirst(idx Index, key cid.Cid) (uint64, error) {
 	return firstOffset, err
 }
 
+// func Has(idx Index, key cid.Cid) (bool, error) {
+// 	has := false
+// 	err := idx.GetAll(key, func(_ uint64) bool {
+//
+// 	})
+// }
+
```
Comment (masih) [inline v2/index/index.go:111]: It seems the scope of this PR expanded since my last comment. Is that intended?

**Review by masih — CHANGES_REQUESTED** [review]

File: v2/index/index.go (hunk 2 above)
Comment (MichaelMure) [reply v2/index/index.go:111]: my fingers decided to type `git commit -a` after I carefully selected my changes ...

**Review by masih — APPROVED** [review]

File: v2/index/index.go (hunk 2 above)
Comment (masih) [reply v2/index/index.go]: no worries thank you

**Comment by MichaelMure** [comment]
FYI I don't have the power to merge.

**Pull request merged**